apps/tx-builder/src/components/Dashboard.tsx

```diff
@@ -53,41 +54,54 @@ const StyledAddressInput = styled(AddressInput)`
 const Dashboard = (): ReactElement => {
   const { web3, interfaceRepo, chainInfo } = useServices();
   const { transactions, handleAddTransaction, handleRemoveTransaction, handleSubmitTransactions } = useTransactions();
-  const [addressOrAbi, setAddressOrAbi] = useState('');
+  const [address, setAddress] = useState('');
+  const [abi, setAbi] = useState('');
   const [isABILoading, setIsABILoading] = useState(false);
   const [contract, setContract] = useState<ContractInterface | null>(null);
   const [loadContractError, setLoadContractError] = useState('');
 
   // Load contract from address or ABI
   useEffect(() => {
-    const loadContract = async (addressOrAbi: string) => {
-      setContract(null);
+    const loadContract = async (address: string) => {
       setLoadContractError('');
 
-      if (!addressOrAbi || !interfaceRepo) {
+      if (!address || !interfaceRepo) {
         return;
       }
 
       try {
-        setIsABILoading(true);
-        const contract = await interfaceRepo.loadAbi(addressOrAbi);
-        setContract(contract);
+        if (isValidAddress(address)) {
+          setIsABILoading(true);
+          setAbi(JSON.stringify(await interfaceRepo.loadAbi(address)));
+        }
       } catch (e) {
-        setContract(null);
+        setAbi('');
         setLoadContractError('No ABI found for this address');
         console.error(e);
       }
       setIsABILoading(false);
     };
 
-    loadContract(addressOrAbi);
-  }, [addressOrAbi, interfaceRepo]);
+    loadContract(address);
+  }, [address, interfaceRepo]);
+
+  useEffect(() => {
+    if (!abi || !interfaceRepo) {
+      setContract(null);
+      return;
+    }
+
+    setContract(interfaceRepo.getMethods(abi));
+  }, [abi, interfaceRepo]);
 
   const getAddressFromDomain = (name: string): Promise<string> => {
     return web3?.eth.ens.getAddress(name) || new Promise((resolve) => resolve(name));
   };
 
-  const isValidAddressOrContract = (isValidAddress(addressOrAbi) || contract) && !isABILoading;
+  const contractHasMethods = contract && contract.methods.length > 0 && !isABILoading;
```

Comment:
I am not 100% sure about adding this extra condition about contract methods length here...

if you use a contract ABI without methods:
```
[
            {
              "inputs": [],
              "stateMutability": "nonpayable",
              "type": "constructor"
            }

]
```
we are not showing the form:

![Captura de pantalla 2022-02-09 a las 11 50 02](https://user-images.githubusercontent.com/26763673/153183351-1bc34495-8652-431b-94bc-79b5644e8856.png)

Reply:
aha. No public methods available. Don't know, the current UI shows a message and the address field but seems as an bug because you cannot do anything with that

Reply:
perhaps we can show a warning

Reply:
The current UI shows only the "To Address" field (And the possibility to use a custom hex data encoded) 

![Captura de pantalla 2022-02-09 a las 11 56 42](https://user-images.githubusercontent.com/26763673/153184519-16f8fbf0-9473-4d0d-b578-ce4cb3596bec.png)

I think that we can go ahead with this new condition.